### PR TITLE
feat(capsule): add background process management tools (#417)

### LINKED ISSUE #417: Background Process Management Tools
`run_shell_command` is strictly synchronous. If the agent tries to start a dev server (e.g., `npm run dev`), the tool hangs until the React loop watchdog times out.

### Tasks
- Add `spawn_background_process`, `read_process_logs`, and `kill_process` tools to `astrid-capsule-shell`.

### PR DESCRIPTION
**Pull request by joshuajbouw**
## Summary

- Add `spawn_background_process`, `read_process_logs`, and `kill_process` WASM host functions and shell capsule tools to solve the synchronous blocking problem when starting long-running processes like `npm run dev`
- Host-side process table in `HostState` with opaque handle IDs (not OS PIDs), bounded stdout/stderr buffers (1MB cap via `VecDeque` ring buffer), and process group kill (`killpg`) to prevent orphaned grandchildren on macOS
- Same sandbox (`SandboxCommand::wrap`), security gate (`check_host_process`), catastrophic-command blocking, and approval flow as existing synchronous `run_shell_command`

## Test Plan

```bash
cargo test --workspace -- --quiet
```

8 unit tests covering:
- Buffer cap enforcement (1MB ring buffer with front-drain)
- `drain_buffer` clear/return and empty-buffer edge case
- `ManagedProcess::Drop` kills child process (verified via `kill(pid, 0)`)
- Process limit enforcement (at-limit rejection and below-limit acceptance)
- `kill_nonexistent_returns_error` (invalid handle lookup)
- `kill_returns_final_output` (buffered stdout survives drain)
- `read_logs_after_natural_exit` (output available after process exits naturally)

### Security review

- P0 fix: `child` wrapped in `Option<Child>` so `Drop` skips `killpg` after explicit kill or `try_wait` reap (prevents killing innocent processes on PID reuse)
- P1 fix: child wrapped in `ManagedProcess` immediately after `spawn()` so early-return errors trigger `Drop` cleanup (no orphan processes)
- P2 fix: cleanup hint in `read_process_logs` output when process has exited (prevents LLM agents from exhausting the 8-slot limit)

## Related Issues

Closes #417

### CODE REVIEW

File: crates/astrid-capsule/src/engine/wasm/host/process.rs [hunk 1]
```diff
@@ -88,3 +96,603 @@ pub(crate) fn astrid_spawn_host_impl(
...
+            if excess > 0 {
+                locked.drain(..excess);
+            }
+        }
+
+        let locked = buf.lock().unwrap_or_else(|e| e.into_inner());
+        assert_eq!(locked.len(), MAX_BUFFER_BYTES);
+        // The oldest 500 bytes should have been dropped.
+        assert_eq!(locked[0], b'A');
+    }
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-capsule/src/engine/wasm/host/process.rs:726]: @gemini-code-assist Fixed - used non-uniform data with marker bytes (X at index 0, Y at index 500) to verify oldest data is specifically dropped. The assertion now checks `locked[0] == b'Y'` and `\!locked.contains(&b'X')`.

File: crates/astrid-capsule-shell/src/lib.rs [hunk 2]
```diff
@@ -359,6 +380,99 @@ impl ShellTools {
...
+        }
+        if !logs.stdout.is_empty() {
+            output.push_str(&format!("--- stdout ---\n{}\n", logs.stdout));
+        }
+        if !logs.stderr.is_empty() {
+            output.push_str(&format!("--- stderr ---\n{}\n", logs.stderr));
+        }
+        if logs.stdout.is_empty() && logs.stderr.is_empty() {
+            output.push_str("(no new output)\n");
+        }
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-capsule-shell/src/lib.rs:449]: @gemini-code-assist Fixed - replaced `format!+push_str` with `write!`/`writeln!` from `std::fmt::Write` to avoid intermediate string allocations.

File: crates/astrid-capsule-shell/src/lib.rs [hunk 3]
```diff
@@ -359,6 +380,99 @@ impl ShellTools {
...
+            None => "unknown exit code".to_string(),
+        };
+
+        let mut output = format!("Process {} killed ({exit_info}).\n", args.id);
+        if !result.stdout.is_empty() {
+            output.push_str(&format!("--- final stdout ---\n{}\n", result.stdout));
+        }
+        if !result.stderr.is_empty() {
+            output.push_str(&format!("--- final stderr ---\n{}\n", result.stderr));
+        }
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-capsule-shell/src/lib.rs:474]: @gemini-code-assist Fixed - same `write!` change applied to kill_process output formatting.

File: crates/astrid-capsule/src/engine/wasm/host/process.rs [hunk 4]
```diff
@@ -88,3 +96,603 @@ pub(crate) fn astrid_spawn_host_impl(
...
+            }
+        }
+
+        // Drain should return the output.
+        let stdout = drain_buffer(&stdout_buf);
+        assert!(
+            stdout.contains("final output"),
+            "expected 'final output' in stdout, got: {stdout}"
+        );
+    }
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-capsule/src/engine/wasm/host/process.rs:893]: @gemini-code-assist Fixed - refactored to use `spawn_reader_thread` instead of duplicating the pipe-read logic. Also removed the now-unused `use std::io::Read as _` import from the test module.
